### DISCUSSION TITLE
Setting height in config file for FeatureTrack and QuantitativeTrack

### DISCUSSION THREAD

**Discussion by anamariaelek** [GMOD/jbrowse-components#3028] [Q&A] [answered]
I can't find a way to define track height, in particular for `FeatureTrack` and `QuantitativeTrack` types. This can be done by manually dragging and resizing the tracks in the browser, but I would want to do it systematically in the configuration.

Sorry if I am missing it somewhere.

Thanks,
Anamaria

**Comment by cmdcolin** [comment, marked as answer]
Hi there,
This is a good question

The answer at the moment is that there is actually not a config variable for the height of a track (or perhaps, for the default height of a track). So, the height is always the same for all tracks, and the resize handle is the only way to change height

On QuantiativeTracks, there are no config track height config variables, and on FeatureTracks it is only "maxHeight" (e.g. the default scroll area height)

I think this is a good idea though, we can convert this into a feature request!

**Reply by cmdcolin** [reply]
Note that there is a "session variable" for this, so you could share a session with your preferred track heights for the tracks that are open in the session ( or similarly create a defaultSession in the config file, which you can use the admin server to help with) but to add it to a bunch of tracks by default, we will want to make a config slot, so this new issue is available to track that https://github.com/GMOD/jbrowse-components/issues/3030

**Reply by anamariaelek** [reply]
Thanks a lot for prompt reply! Could you save me some time digging through the examples trying to find which parameter I need to set in json file to change the height of all session tracks?

**Reply by cmdcolin** [reply]
sure. here is for example, the result of running File->Export session, and inspecting the JSON

```
{
  "session": {
    "id": "iXRMq5E_4",
    "name": "Integration test example 6/13/2022, 10:23:50 AM",
    "margin": 0,
    "drawerWidth": 384,
    "views": [
      {
        "id": "integration_test",
        "type": "LinearGenomeView",
        "offsetPx": 2000,
        "bpPerPx": 0.05,
        "displayedRegions": [
          {
            "refName": "ctgA",
            "start": 0,
            "end": 50001,
            "reversed": false,
            "assemblyName": "volvox"
          }
        ],
        "tracks": [
          {
            "id": "PTcKa0RfU",
            "type": "QuantitativeTrack",
            "configuration": "volvox_microarray",
            "displays": [
              {
                "id": "0vRpZdzVdT",
                "type": "LinearWiggleDisplay",
                "height": 303,
                "configuration": "volvox_microarray-LinearWiggleDisplay",
                "selectedRendering": "",
                "resolution": 1,
                "constraints": {}
              }
            ]
          },
          {
            "id": "JGUoMiAJ7",
            "type": "VariantTrack",
            "configuration": "volvox_filtered_vcf",
            "displays": [
              {
                "id": "BieZjAuFBB",
                "type": "LinearVariantDisplay",
                "height": 331,
                "configuration": "volvox_filtered_vcf-LinearVariantDisplay"
              }
            ]
          }
        ],
        "hideHeader": false,
        "hideHeaderOverview": false,
        "trackSelectorType": "hierarchical",
        "trackLabels": "offset",
        "showCenterLine": false,
        "showCytobandsSetting": true
      }
    ],
    "widgets": {
      "hierarchicalTrackSelector": {
        "id": "hierarchicalTrackSelector",
        "type": "HierarchicalTrackSelectorWidget",
        "collapsed": {
          "volvox_microarray": false,
          "volvox_filtered_vcf": false
        },
        "filterText": "",
        "view": "integration_test"
      }
    },
    "activeWidgets": {
      "hierarchicalTrackSelector": "hierarchicalTrackSelector"
    },
    "connectionInstances": [],
    "sessionTracks": [],
    "sessionConnections": [],
    "sessionAssemblies": [],
    "sessionPlugins": [],
    "minimized": false,
    "drawerPosition": "right"
  }
}
```

you can copy this whole thing and put it onto a "defaultSession" key in your config.json if you wanted it to always load that particular session. the session json is a bit verbose, but a lot are default values that can be removed if you wanted to make a more minimal session json

the key for track height in particular is the height on the "display" part of the session (that governs how the track is displayed in a given view)

```
                "height": 303, 
                "configuration": "volvox_microarray-LinearWiggleDisplay",
```

and 


```
                "height": 331, 
                "configuration": "volvox_filtered_vcf-LinearVariantDisplay"
```

I used the resize handle to resize those two tracks after opening them. note that the height here is only available for tracks that are open (tracks that are closed are not part of the session)

**Reply by anamariaelek** [reply]
Thanks a lot, the example is very helpful.
